index.js

```diff
@@ -46,13 +47,22 @@ module.exports = function(plugins, options) {
             };
 
             if (node.tag === 'style' && node.content) {
-                var styles = indentResolve([].concat(node.content).join(''), indent);
-                promise = css.process(styles, options)
-                    .then(function(result) {
-                        node.content = [indentResolve(result.css, indent)];
-                    });
+                var includesStandardType = filterType.test('text/css');
+                var hasTypeAttr = node.attrs && node.attrs.type;
+                var isFilterType = hasTypeAttr && filterType.test(node.attrs.type.trim());
+                var isUndefinedType = !hasTypeAttr || includesStandardType && /^(\s+)?$/.test(node.attrs.type);
+                var isStandardCSS = includesStandardType && (isUndefinedType || isFilterType);
+                var isOther = !isStandardCSS && isFilterType;
 
-                promises.push(promise);
+                if (isStandardCSS || isOther) {
```

Comment:
Basically ok :) but it can be implemented more straight forward imho

@stevenvachon Do you have any specific usecase in terms of MIME types or do you just need the ability to filter certain nodes ?